src/main/scala/circt/stage/phases/CIRCT.scala

```diff
@@ -185,55 +217,47 @@ class CIRCT extends Phase {
             )
         })
 
-    try {
-      logger.info(s"""Running CIRCT: '${cmd.mkString(" ")} < $$input'""")
-      println(s"""Running CIRCT: '${cmd.mkString(" ")} < $$input'""")
-      val stdoutStream, stderrStream = new java.io.ByteArrayOutputStream
-      val stdoutWriter = new java.io.PrintWriter(stdoutStream)
-      val stderrWriter = new java.io.PrintWriter(stderrStream)
-      val exitValue = (cmd #< new java.io.ByteArrayInputStream(input.getBytes))
-        .!(ProcessLogger(stdoutWriter.println, stderrWriter.println))
-      stdoutWriter.close()
-      stderrWriter.close()
-      val result = stdoutStream.toString
+    logger.info(s"""Running CIRCT: '${cmd.mkString(" ")} < $$input'""")
+    println(s"""Running CIRCT: '${cmd.mkString(" ")} < $$input'""")
+    val stdoutStream, stderrStream = new java.io.ByteArrayOutputStream
+    val stdoutWriter = new java.io.PrintWriter(stdoutStream)
+    val stderrWriter = new java.io.PrintWriter(stderrStream)
+    val exitValue = (cmd #< new java.io.ByteArrayInputStream(input.getBytes))
+      .!(ProcessLogger(stdoutWriter.println, stderrWriter.println))
+    stdoutWriter.close()
+    stderrWriter.close()
+    val result = stdoutStream.toString
+    logger.info(result)
+    val errors = stderrStream.toString
+    if (exitValue != 0)
```

Comment:
In some environments a missing firtool binary led to an exception here instead of a nonzero exit code

Have you tested what this does if firtool is not on the path?

Reply:
This appears to not have been working correctly before either. I added a commit to do the right thing here.

This is the current output with the new `FirtoolNotFound` exception. The `Process` that gets created seems to do its own thing with exception printing that I can't sort out how to control. This will now narrowly try/catch the `Process` to sort out what happened if it returns a `RunTimeException` with a message indicating that firtool wasn't found:

```
Running CIRCT: 'firtool -format=fir -warn-on-unprocessed-annotations -verify-each=false -dedup -output-annotation-file circt.anno.json -disable-infer-rw -annotation-file Foo.anno.json < $input'
Exception in thread "Thread-9668" java.io.IOException: Cannot run program "firtool": error=2, No such file or directory
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
	at scala.sys.process.ProcessBuilderImpl$Simple.run(ProcessBuilderImpl.scala:75)
	at scala.sys.process.ProcessImpl$PipedProcesses.runAndExitValue(ProcessImpl.scala:142)
	at scala.sys.process.ProcessImpl$PipedProcesses.runAndExitValue(ProcessImpl.scala:121)
	at scala.sys.process.ProcessImpl$CompoundProcess.$anonfun$x$4$1(ProcessImpl.scala:100)
	at scala.sys.process.ProcessImpl$Spawn$$anon$1.run(ProcessImpl.scala:27)
Caused by: java.io.IOException: error=2, No such file or directory
	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:340)
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:271)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1107)
	... 6 more
circt.stage.phases.Exceptions$FirtoolNotFound: firtool not found
------------------------------------------------------------------------------
Chisel requires that firtool, the MLIR-based FIRRTL Compiler (MFC), is installed
and available on your $PATH.  (Did you forget to install it?)  You can download
a binary release of firtool from the CIRCT releases webpage:
  https://github.com/llvm/circt/releases
------------------------------------------------------------------------------
  at circt.stage.phases.CIRCT.transform(CIRCT.scala:247)
  at circt.stage.phases.CIRCT.transform(CIRCT.scala:117)
  at firrtl.options.DependencyManager.$anonfun$transform$5(DependencyManager.scala:280)
  at firrtl.Utils$.time(Utils.scala:181)
  at firrtl.options.DependencyManager.$anonfun$transform$3(DependencyManager.scala:280)
  at scala.collection.LinearSeqOptimized.foldLeft(LinearSeqOptimized.scala:126)
  at scala.collection.LinearSeqOptimized.foldLeft$(LinearSeqOptimized.scala:122)
  at scala.collection.immutable.List.foldLeft(List.scala:91)
  at firrtl.options.DependencyManager.transform(DependencyManager.scala:269)
  at firrtl.options.DependencyManager.transform$(DependencyManager.scala:255)
  at firrtl.options.PhaseManager.transform(DependencyManager.scala:443)
  at circt.stage.ChiselStage$.emitSystemVerilog(ChiselStage.scala:108)
  ... 55 elided
```

Reply:
Thanks!